lib/run_part.c

```diff
@@ -15,23 +16,23 @@
 #include "shadowlog_internal.h"
 
 
-static int run_part(char *script_path, const char *name, const char *action)
+static int run_part(int script_fd, const char *script_name, const char *name, const char *action)
 {
 	pid_t pid;
 	int wait_status;
 	pid_t pid_status;
-	char *args[] = { script_path, NULL };
+	char *args[] = { (char *) script_name, NULL };
 
 	pid=fork();
 	if (pid==-1) {
 		fprintf(shadow_logfd, "fork: %s\n", strerror(errno));
 		return 1;
 	}
 	if (pid==0) {
-		setenv("ACTION",action,1);
-		setenv("SUBJECT",name,1);
-		execv(script_path,args);
-		fprintf(shadow_logfd, "execv: %s\n", strerror(errno));
+		setenv ("ACTION",action,1);
+		setenv ("SUBJECT",name,1);
+		fexecve (script_fd, args, environ);
+		fprintf(shadow_logfd, "failed to execute %s: %s\n", script_name, strerror(errno));
```

Comment:
IMHO, `"fexecve(%s): %s"` would be better.

Reply:
Changed.